DSpace/dspace-angular
# Made it possible to enable video viewer without also enabling image viewer

## References
* Fixes #2035

## Description
Made it possible to only enable the video player without also enabling image viewer and found a few other related bugs and fixed them too.

## Instructions for Reviewers
List of changes in this PR:
* Show the `MediaViewerComponent` on item pages when one of the `mediaViewer` config options is set to true instead of only checking if `image` is set to `true`.
* When the first `ORIGINAL` bitstream is a non-supported file (like an `*.xml` file) or a supported file who is currently disabled (for example when `image: false` and that the first `ORIGINAL` bitstream is a picture), the media viewer doesn't work.
* The thumbnails weren't always rendered because in the `MediaViewerImageComponent`, the `galleryImages` is only initiated once in `ngOnInit()` instead of `ngOnChanges()`, which means that sometimes when the `thumbnailsRD$` from `MediaViewerComponent` isn't initialised before `MediaViewerImageComponent.ngOnInit()` is run the default fallback is used and will not be replaced by the thumbnail when `thumbnailsRD$` is initialised.
* When there are 0 `ORIGINAL` bitstreams but there is a `THUMBNAIL` bitstream the `THUMBNAIL` would never be shown because it always renders the thumbnail from the element in `mediaList`.
* The `Subscription`s were not properly cleaned up after the `MediaViewerComponent` is destroyed.

**Guidance for how to test or review this PR:**
* Set the `image` & `video` config options from `mediaViewer` both to `false`, both to `true` and try it once by disabling one but leaving the other enabled:
  * Create a few new items with only image(s), only video(s)/audio file(s) and only non-image/audio/video file§s). Check that fore each of those items the correct placeholder is rendered.
  * Add `THUMBNAIL` bitstreams and check that they are correctly rendered.
  * Check that you can still use the [video captioning](https://wiki.lyrasis.org/display/DSDOC7x/User+Interface+Configuration#UserInterfaceConfiguration-MediaViewerSettings) feature.

## Checklist
